test/users/engin/compiler/innerRecord.future

```diff
@@ -0,0 +1,8 @@
+bug: nil dereference referencing outer class' fields
+
+NOTES
```

Comment:
The problem seems to be that innerRecord's "outer" is not getting set properly during normal record initialization, so attempts to access one of outer's fields results in a dereference of nil.

I would call out in this future that calling the record's initializer via new in the field declaration allows the program to compile fine. (It's good that you put it in the file itself, I just think it should be in here as well)